monetio/sat/_mopitt_l3_mm.py

```diff
@@ -0,0 +1,147 @@
+    # print("READING FILE " + inFileName)
+    # fName = os.path.basename(filename)
+
+    try:
+        inFile = h5py.File(filename, "r")
+    except Exception:
+        print("ERROR: CANNOT OPEN " + filename)
+        return 0
+
+    grp = inFile[structure]
+    k = grp.attrs
+    startTimeBytes = k.get("StartTime", default=None)
+    startTime = pd.to_datetime(startTimeBytes[0], unit="s", origin="1993-01-01 00:00:00")
+    # print("******************", startTime)
+
+    try:
+        inFile.close()
+    except Exception:
+        print("ERROR CANNOT CLOSE " + filename)
+        return 0
+
+    return startTime
+
+
+def loadAndExtractGriddedHDF(filename, varname):
+    """Method to open MOPITT gridded hdf files.
+    Masks data that is missing (turns into np.nan).
+
+    Parameters
+    ----------
+    filename : str
+        Path to the file.
+    varname : str
+        The variable to load from the MOPITT file.
+
+    Returns
+    -------
+    xarray.Dataset
+    """
+    import h5py
+
+    # initialize into dataset
+    ds = xr.Dataset()
+
+    # load the dimensions
+    he5_load = h5py.File(filename, mode="r")
+    lat = he5_load["/HDFEOS/GRIDS/MOP03/Data Fields/Latitude"][:]
+    lon = he5_load["/HDFEOS/GRIDS/MOP03/Data Fields/Longitude"][:]
+    alt = he5_load["/HDFEOS/GRIDS/MOP03/Data Fields/Pressure2"][:]
+    alt_short = he5_load["/HDFEOS/GRIDS/MOP03/Data Fields/Pressure"][:]
+
+    # 2D or 3D variables to choose from
+    variable_dict = {
+        "column": "/HDFEOS/GRIDS/MOP03/Data Fields/RetrievedCOTotalColumnDay",
+        "apriori_col": "/HDFEOS/GRIDS/MOP03/Data Fields/APrioriCOTotalColumnDay",
+        "apriori_surf": "/HDFEOS/GRIDS/MOP03/Data Fields/APrioriCOSurfaceMixingRatioDay",
+        "pressure_surf": "/HDFEOS/GRIDS/MOP03/Data Fields/SurfacePressureDay",
+        "ak_col": "/HDFEOS/GRIDS/MOP03/Data Fields/TotalColumnAveragingKernelDay",
+        "ak_prof": "/HDFEOS/GRIDS/MOP03/Data Fields/APrioriCOMixingRatioProfileDay",
+    }
+    try:
+        data_loaded = he5_load[variable_dict[varname]][:]
+    except Exception:
+        print("ERROR: Cannot load " + varname + " from " + filename)
+        return 0
+
+    he5_load.close()
+
+    # DEBEG
+    # print(data_loaded.shape)
+
+    # create xarray DataArray
+    if (
+        varname == "column"
+        or varname == "apriori_col"
+        or varname == "apriori_surf"
+        or varname == "pressure_surf"
+    ):
+        ds[varname] = xr.DataArray(data_loaded, dims=["lon", "lat"], coords=[lon, lat])
+        # missing value -> nan
+        ds[varname] = ds[varname].where(ds[varname] != -9999.0)
+    elif varname == "ak_col":
+        ds[varname] = xr.DataArray(data_loaded, dims=["lon", "lat", "alt"], coords=[lon, lat, alt])
+    elif varname == "apriori_prof":
+        ds[varname] = xr.DataArray(
+            data_loaded, dims=["lon", "lat", "alt"], coords=[lon, lat, alt_short]
+        )
+
+    return ds
+
+
+def read_mopittdataset(files, varname):
```

Comment:
```suggestion
def open_dataset(files, varname):
```
More general name. The module it is in still identifies it to be for MOPITT.